### PR TITLE
Add markdownlint

### PR DESCRIPTION
**Pull request by m1cr0man**

### CODE REVIEW

File: .github/workflows/build-validate.yml [hunk 1]
```diff
@@ -17,17 +14,19 @@ jobs:
...
     - name: Install dependencies
       run: |
+        gem install html-proofer mdl &
         python -m pip install --upgrade pip
-        pip install -r requirements.txt &
-        gem install html-proofer
+        pip install -r requirements.txt
         wait
+    - name: Run markdownlint (mdl)
+      run: mdl .
```
Comment (butlerx) [inline .github/workflows/build-validate.yml]: `mdl . -c .mdlrc` 
Use the mdlrc file from before

**Comment by m1cr0man** [comment]
There was a lot of linting errors ;)

File: docs/hardware/ipkvm.md [hunk 2]
```diff
@@ -14,31 +14,30 @@ as we have v1 of the dongles and not v2.
...
 ## Connecting remotely
 
 You need a proprietary tool from Dell to connect remotely. There are two versions available.
 
 - The latest version (v4) uses HTTP to connect to the remote screen. It works on Linux but has an iffy
-Java based viewer. Download it [here](http://downloads.dell.com/RACK%20SOLUTIONS/DELL_MULTI-DEVICE_A02_R270943.exe).
+  Java based viewer. Download it [here](http://downloads.dell.com/RACK%20SOLUTIONS/DELL_MULTI-DEVICE_A02_R270943.exe).
 - The older version (v3) uses weird ports only to connect to the remote screen. It has no Linux support
-but works absolutely fine in Wine, which can be easier than the Linux install. Download it [here](http://downloads.dell.com/RACK%20SOLUTIONS/R132099.EXE)
+  but works absolutely fine in Wine, which can be easier than the Linux install. Download it [here](http://downloads.dell.com/RACK%20SOLUTIONS/R132099.EXE)
```
Comment (butlerx) [inline docs/hardware/ipkvm.md]: Can we make this https?

File: docs/monitoring/collectd.md [hunk 3]
```diff
@@ -1,29 +1,29 @@
 # Collectd
 
 - Full man page for collectd with full descriptions and configs:
-  - ![Collectd.conf(5)](https://collectd.org/documentation/manpages/collectd.conf.5.shtml)
+    - ![Collectd.conf(5)](https://collectd.org/documentation/manpages/collectd.conf.5.shtml)
 - Collectd is installed on ubuntu and bsd machines
-  - Metrics are written to `zeus.internal` and then scrped by prometheus
+    - Metrics are written to `zeus.internal` and then scrped by prometheus
```
Comment (butlerx) [inline docs/monitoring/collectd.md]: Scraped

File: docs/plans/TODO-2017.md [hunk 4]
```diff
@@ -8,8 +8,8 @@
 - [x] move audited scripts from /srv/admin/scripts and /srv/admin/scrips/rrs to
       github.
 - [ ] version control important configs
-  - [ ] DNS
-  - [ ] apache
+    - [ ] DNS
```
Comment (butlerx) [inline docs/plans/TODO-2017.md]: Can we just dump this file

**Comment by m1cr0man** [comment]
Fixed all those things

File: .markdown.style.rb [hunk 5]
```diff
@@ -2,5 +2,6 @@
 rule 'MD013', :line_length => 100, :code_blocks => false, :tables => false
 rule 'MD026', :punctuation => '.,;:'
 rule 'MD007', :indent => 4
+exclude_rule 'MD033'
```
Comment (butlerx) [inline .markdown.style.rb:5]: out of curiosity where do we inline html?

File: docs/procedures/nixos.md [hunk 6]
```diff
@@ -3,36 +3,50 @@
...
+if you have access a branch is appropriate, in all other cases make a fork and pr back to
+Redbrick's repo. This will be case by case for those of you reading.
 
- - https://github.com/redbrick/nix-configs
- 
-Depending on the powers that be, some sort of normal pr contribution will be acceptable, if you have access a branch is appropriate, in all other cases make a fork and pr back to redbricks repo. This will be case by case for those of you reading
+Here's a quick hit list of stuff that's worthy of book marking also as you work with Nix:
 
-Here's a quick hitlist of stuff that's worthy of book marking also as you work with nix
+- [NixOS Wiki](https://nixos.wiki/wiki/Main_Page)
```
Comment (butlerx) [inline docs/procedures/nixos.md:29]: :heart:

**Review by butlerx — APPROVED** [review]

**Pull request merged**